daemon/redis.c

```diff
@@ -283,13 +280,22 @@ void onRedisNotification(redisAsyncContext *actx, void *reply, void *privdata) {
 	if (rr->elements != 4)
 		goto err;
 
-	char *pch = strstr(rr->element[2]->str, "notifier-");
-	if (pch == NULL) {
-		rlog(LOG_ERROR,"Redis-Notifier: The substring 'notifier-' has not been found in the redis notification !\n");
-		goto err;
+	char *pch =0;
+	if (cm->conf.redis_multikey) {
+		pch = strstr(rr->element[2]->str, "notifier-");
+		if (pch == NULL) {
+			rlog(LOG_ERROR,"Redis-Notifier: The prefix 'notifier-' to determine the redis key has not been found in the redis notification !\n");
+			goto err;
+		}
+		pch += strlen("notifier-");
+	} else {
+		pch = strstr(rr->element[2]->str, "json-");
+		if (pch == NULL) {
+			rlog(LOG_ERROR,"Redis-Notifier: The prefix 'json-' to determine the redis key has not been found in the redis notification !\n");
+			goto err;
+		}
```

Comment:
Yes right, it's not missing and it's not very nice but will be replaced when I removed the old multikey behaviour in a subsequent change (I am already working on it)